### ISSUE TITLE
Couple issues with Dynamic Routing with MVC Page Templates

### ISSUE THREAD

**Issue by liamgold** [KenticoDevTrev/DynamicRouting#18]
When registering a dynamic route, and registering a page template for a controller:
```
[assembly: RegisterPageTemplate("Example.DetailTemplate", typeof(DetailController), "Example template")]
[assembly: DynamicRouting(typeof(DetailController), new string[] { ExampleDetail.CLASS_NAME }, nameof(DetailController.Index))]
```
If you don't specify the full path to a view, e.g.
```
        public ActionResult Index()
        {
            var page = DynamicRouteHelper.GetPage<ExampleDetail>();

            ...

            // Render the view.
            return View(model); <--- this needs to be a full path, e.g. return View("~/Views/Detail/Index.cshtml", model);
        }
```
It errors, because it's not checking the right folder:
```
Message: The view 'Index' or its master was not found or no view engine supports the searched locations. The following locations were searched:
~/Views/DynamicRouteTemplate/Index.aspx
~/Views/DynamicRouteTemplate/Index.ascx
~/Views/Shared/Index.aspx
~/Views/Shared/Index.ascx
~/Views/DynamicRouteTemplate/Index.cshtml
~/Views/DynamicRouteTemplate/Index.vbhtml
~/Views/Shared/Index.cshtml
~/Views/Shared/Index.vbhtml
```
Is there a way of doing this without having to specify the full path?

---

In the same scenario, rendering a partial view (which lives in a subfolder `~/Views/Detail/Partials/TestArea.cshtml`) from the view, usually you can render the partial like:
```
@Html.Partial("Partials/TestArea", Model)
```
But when using MVC page templates with the dynamic routing, the same thing happens and I need to specify the full path, e.g.:
```
@Html.Partial("~/Views/Detail/Partials/TestArea.cshtml", Model)
```
Is there a way of doing this without having to specify the full path?

**Comment by KenticoDevTrev** [comment]
This is odd, it doesn't behave this way for me, I have a unit controller that is sent there via dynamic routing, I just call return view (model) and it finds my views/unit/index.cshtml 

Do you have time for a screen share that I can look okay your instance? Email me at tfayas@hbs.net with your availability today. I'm in central time zone

**Comment by KenticoDevTrev** [comment]
Then again, I'm guessing your using the page templates approach right? I'll do some digging, I know if you don't do a page templates it works normally.

**Comment by liamgold** [comment]
Yeah I'm trying to use MVC page templates along with the routing, it had been working fine before introducing MVC page templates.  The controllers that don't have the page templates enabled work fine too.

I'm probably going to have a play around with this later in the evening (UK time), might drop you an email soon.

Also noticed when using MVC page templates that the routing was always looking for the Index action rather than the one specified in the attribute, e.g. Test:
```
[assembly: DynamicRouting(typeof(DetailController), new string[] { ExampleDetail.CLASS_NAME }, nameof(DetailController.Test))]
```

**Comment by KenticoDevTrev** [comment]
Fixed with commit 2101a1eb8cc114461974d9acda13caa31f2309af , thanks a bunch @liamgold , the new nuget package is uploaded (12.29.10), may take just a little bit of time to show up!
